workspaces/tests/create_account.rs

```diff
@@ -0,0 +1,20 @@
+use workspaces::prelude::*;
+
+#[tokio::test]
+async fn test_subaccount_creation() -> anyhow::Result<()> {
+    let worker = workspaces::sandbox();
+    let account = worker.dev_create().await?;
```

Comment:
`dev_create` seems like less than ideal name. Dev create what? Maybe `worker.test_account()` or `worker.new_test_account()`? `worker.test_account()` which always returns *the same* account sounds like a potentially good idea.

Reply:
so I wanted to tie this with `dev_deploy` since they're very similar except `dev_deploy` also deploys a contract. How about `dev_create_account()`? Returning the same account will be problematic for testnet since that could cause some non-determinism if every testnet test used the same account and modified the accounts state

Reply:
>Returning the same account will be problematic for testnet since that could cause some non-determinism if every testnet test used the same account

I mean that "the same" is scoped to the `worker` -- basically, "create a new dev account" is something you use rarely, and for most tests, which need just a single account, you use "get the test account", which is created on the first access. 

Not sure whether this is a good idea...

Thinking more about this, it's probably a good idea -- for the first release we should aim for orthogonal API, and we can add shortcuts later.

Reply:
Potentially horrible idea this is async function, so the user already has to type `.await`. So what we *can* do is

```rust
let account_without_code = worker.dev_create_account().await?;
let account_with_code = worker.dev_create_account().with_code(&wasm).await?;
```

This reads pretty nice at the call-site, but would be quite mind bending in the rustdoc. I wonder if there's some precedent somewhere for such async builders?

Reply:
> I mean that "the same" is scoped to the worker -- basically, "create a new dev account" is something you use rarely, and for most tests, which need just a single account, you use "get the test account", which is created on the first access.

That pretty much is the `root_account` for Sandbox which covers most of those cases. For Testnet we can emulate this by doing what you describe. 

> Thinking more about this, it's probably a good idea -- for the first release we should aim for orthogonal API, and we can add shortcuts later.

Wait, did you mean bad idea instead of good idea here based on you saying we should keep it separate or orthogonal.

> I wonder if there's some precedent somewhere for such async builders?

hmm, yeah's let consider this for the release after the first one I'm making today or tomorrow. Would be interesting if we made the builder `await`able as that could replace the `transact()` calls